common/persistence/sql/sqlHistoryStore.go

```diff
@@ -117,7 +117,8 @@ func (m *sqlHistoryStore) AppendHistoryNodes(
 			DataEncoding: string(blob.Encoding),
 		}
 
-		return m.txExecute(ctx, "AppendHistoryNodes", func(tx sqlplugin.Tx) error {
+		dbShardID := sqlplugin.GetDBShardIDFromHistoryShardID(request.ShardID, m.db.GetTotalNumDBShards())
```

Comment:
This will cause large partition issue. We originally used `shard_id` as the partition key of sharded SQL, but the latency of read history operation kept growing as the bench test ran.
When you do continue as new, all workflows with the same workflow id will be stored in the same shard. So instead, we use `shard_id, tree_id` as the partition key

Reply:
Yeah good point. Will change it.

Reply:
Changed to GetDBShardIDFromTreeID :D thanks for pointing it out.

Reply:
I think you didn't change all of them?